src/js/providers/html5.js

```diff
@@ -244,6 +253,31 @@ function VideoProvider(_playerId, _playerConfig) {
         }
     }
 
+    function _checkForBuffering(position) {
+        const bufferedRange = _videotag.buffered;
+        if (bufferedRange.length <= 0) {
+            return;
+        }
+        let withinBuffer;
+        if (position <= bufferedRange.end(bufferedRange.length - 1)) {
+            for (let i = 0; i < bufferedRange.length; i++) {
+                withinBuffer = position > bufferedRange.start(i) && position < bufferedRange.end(i);
+                if (withinBuffer) {
+                    break;
+                }
+            }
+        }
+
+        _videotag.removeEventListener('waiting', setBufferingState);
+        if (!withinBuffer) {
+            _videotag.addEventListener('waiting', setBufferingState);
+        }
+    }
+
+    function setBufferingState() {
+        _this.setState(STATE_BUFFERING);
+    }
+
     function _setPositionBeforeSeek(position) {
```

Comment:
This doesn't make sense as a function - it's only called in one place. Just update `_position` or `_positionBeforeSeek`  in `timeupdate`.

Reply:
it's called in pause, seeking, seek and time update